pkg/deploy/reaper/reaper.go

```diff
@@ -57,10 +56,16 @@ func (reaper *DeploymentConfigReaper) Stop(namespace, name string, gracePeriod *
 		// so we shouldn't error out here
 		glog.V(2).Info(err)
 	}
-	if err = reaper.kc.ReplicationControllers(namespace).Delete(util.LatestDeploymentNameForConfig(dc)); err != nil {
-		// The deploymentConfig may not have a replication controller to delete
-		// so we shouldn't error out here
-		glog.V(2).Info(err)
+	rcList, err := reaper.kc.ReplicationControllers(namespace).List(ConfigSelector(dc))
+	if err != nil {
+		return "", err
+	}
+	// Remove all the deployments of the configuration
+	for _, rc := range rcList.Items {
+		if err = reaper.kc.ReplicationControllers(rc.Namespace).Delete(rc.Name); err != nil {
```

Comment:
We have to reap the old arcs as well.  Every single deployment needs to go through scale 0